### PR TITLE
feature(encoder): Implement Iterator over DaBlob for EncodedData

### PR DESCRIPTION
**Pull request by AlejandroCabeza**
## 1. What does this PR implement?
Implement Iterator over DaBlob for EncodedData
Closes: https://github.com/logos-co/nomos-node/issues/943

## 2. Does the code have enough context to be clearly understood?
Yes

## 3. Who are the specification authors and who is accountable for this PR?
@AlejandroCabeza 

## 4. Is the specification accurate and complete?
N/A

## 5. Does the implementation introduce changes in the specification?
N/A

## Checklist

> [!WARNING]  
> Do not merge the PR if any of the following is missing:

* [X] 1. Description added.
* [X] 2. Context and links to Specification document(s) added.
* [X] 3. Main contact(s) (developers and specification authors) added
* [X] 4. Implementation and Specification are 100% in sync including changes. This is critical.
* [X] 5. Link PR to a specific milestone.

### CODE REVIEW

File: nomos-da/kzgrs-backend/src/encoder.rs [hunk 1]
```diff
@@ -55,6 +56,65 @@ pub struct EncodedData {
     pub column_commitments: Vec<Commitment>,
     pub aggregated_column_commitment: Commitment,
     pub aggregated_column_proofs: Vec<Proof>,
+
+    /// Iterator State
+    current_column_idx: Option<usize>,
```
Comment (danielSanchezQ) [inline nomos-da/kzgrs-backend/src/encoder.rs:61]: This is not the way to go. Usually what you want to do is to implement `IntoIterator` then return a new struct that handles this state and have a reference to the original Encoded data and that struct implements `Iterator`.

**Review by danielSanchezQ — CHANGES_REQUESTED** [review]
Implementation looks ok, but having an iteration state inside that structure is wrong. What happens if I need 5 iterators over it? You will need to add 5 states then...

File: nomos-da/kzgrs-backend/src/encoder.rs (hunk 1 above)
Comment (AlejandroCabeza) [reply nomos-da/kzgrs-backend/src/encoder.rs:61]: Oh, right! This actually makes waaaay more sense. I didn't have a great feeling about it, but as everyone was suggesting doing it this way I assumed it was the standard.

Thanks for pointing this out! Updating

**Review by danielSanchezQ — APPROVED** [review]
Looking good, thanks!
We can use this in the executor now. Probably something to do when merging this. Please add an issue and attach to the ongoing iteration.

**Pull request merged**